.github/workflows/1_pre-Test-Model-Image-Build.yaml

```diff
@@ -0,0 +1,88 @@
+name: Test Model Image Build
+run-name: (2_pre) Test Model Image - "${{ github.event.head_commit.message }}"
+
+# This GitHub Actions workflow builds a Docker image for the
+# cfa-epinow2-pipeline-docker project. In-container tests can be added here.
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+  schedule:
+    # Tuesdays at noon GMT time, 24 hours before the prod container is built and deployed
+    - cron: "0 12 * * 2"
+
+env:
+  # Together, these form: cfaprdbatchcr.azurecr.io/cfa-epinow2-pipeline
+  REGISTRY: cfaprdbatchcr.azurecr.io
+  IMAGE_NAME: cfa-epinow2-pipeline
+
+jobs:
+
+  Job01-build_image_dependencies:
+    runs-on: cfa-cdcgov # VM based runner serving CFA's cdcgov repos (as opposed to cdcent)
+
+    outputs:
+      tag: ${{ steps.image-tag.outputs.tag }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Figure out tag (latest if push to main, otherwise pr-<number>)
+        id: image-tag
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "tag=pr-${{ github.event.number }}" >> $GITHUB_OUTPUT
+          else
+            echo "tag=latest" >> $GITHUB_OUTPUT
+          fi
```

Comment:
I'm realizing that this may not work well in the `workflow_dispatch` case and perhaps the schedule case. Based on a little googling, it looks like the `event_name`  for workflow dispatch is `workflow_dispatch` and for schedule is `schedule`. In this if/else, both of these cases will default to updating `latest`. 

The schedule case may not be an issue, but my quick read is that the schedule might run the PRs in addition to main?

In any case, I think this needs some reworking to be [an explicit check](https://stackoverflow.com/questions/58033366/how-to-get-the-current-branch-within-github-actions) whether the the branch is main?

Reply:
I set it up so runs from main use `latest`; otherwise they will use the branch-name.